reference/docs-conceptual/PSScriptAnalyzer/create-custom-rule.md

```diff
@@ -0,0 +1,241 @@
+---
+description: This article provides a basic guide for creating your own customized rules.
+ms.date: 03/22/2022
+title: Creating custom rules
+---
+# Creating custom rules
+
+PSScriptAnalyzer uses the [Managed Extensibility Framework (MEF)](/dotnet/framework/mef/) to import
+all rules defined in the assembly. It can also consume rules written in PowerShell scripts.
+
+When calling `Invoke-ScriptAnalyzer`, users can specify custom rules using the
+**CustomizedRulePath** parameter.
+
+This article provides a basic guide for creating your own customized rules.
+
+## Basic requirements
+
+### Functions should have comment-based help
+
+Include the `.DESCRIPTION` field. This becomes the description for the customized rule.
+
+```powershell
+<#
+.SYNOPSIS
+    Name of your rule.
+.DESCRIPTION
+    This would be the description of your rule. Please refer to Rule Documentation
+    for consistent rule messages.
+.EXAMPLE
+.INPUTS
+.OUTPUTS
+.NOTES
+#>
+```
+
+### Output type should be **DiagnosticRecord**
+
+```powershell
+[OutputType([Microsoft.Windows.PowerShell.ScriptAnalyzer.Generic.DiagnosticRecord[]])]
+```
+
+### Each function must have a Token array or an Ast parameter
+
+The name of the _Ast_ parameter name must end with **Ast**.
```

Comment:
Nit: Not sure if this should be bolded as it _seems_ like an exception to standard formatting. Applies to all other instances of `_parameter_`.
```suggestion
The name of the **Ast** parameter name must end with **Ast**.
```